### PR TITLE
Added Double Circular Linked List

### PR DESCRIPTION
**Pull request by Aditya-y9**
Honourable maintainers,
Please add Hacktoberfest and Hacktoberfest-accepted labels.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/datastructures/lists/DoubleCircularLinkedList [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/**
```
Comment (susanhsrestha) [inline src/main/java/com/thealgorithms/datastructures/lists/DoubleCircularLinkedList:1]: The Java file extension is missing, and the package naming convention is also not provided.

**Review by susanhsrestha — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/datastructures/lists/DoubleCircularLinkedList (hunk 1 above)
Comment (Aditya-y9) [reply src/main/java/com/thealgorithms/datastructures/lists/DoubleCircularLinkedList]: Sorry for that silly mistake i just overlooked it.
It has been fixed now.
Thanks!

**Comment by Aditya-y9** [comment]
@susanhsrestha @BurkHufnagel @arindam89 @qckzr 
Done with all checks
Added Java file extension
Added package accordingly
Please Add Hacktoberfest and Hacktoberfest-accepted labels

Thanks!

**Comment by susanhsrestha** [comment]
All I see is an empty file.

**Review by susanhsrestha — APPROVED** [review]
LGTM!

**Pull request closed without merging**